test/integration/test_dataplane.go

```diff
@@ -204,29 +231,86 @@ func TestDataPlaneEssentials(t *testing.T) {
 
 	t.Log("verifying dataplane status is properly filled with backing service name and its addresses")
 	require.Eventually(t, testutils.DataPlaneHasServiceAndAddressesInStatus(t, GetCtx(), dataplaneName, clients), waitTime, tickTime)
+}
 
-	t.Log("updating dataplane spec with proxy service type of ClusterIP")
-	require.Eventually(t,
-		testutils.DataPlaneUpdateEventually(t, GetCtx(), dataplaneName, clients, func(dp *operatorv1beta1.DataPlane) {
-			dp.Spec.Network.Services.Ingress.Type = corev1.ServiceTypeClusterIP
-		}),
-		waitTime, tickTime)
+func TestDataPlaneServiceTypes(t *testing.T) {
+	t.Parallel()
+	namespace, cleaner := helpers.SetupTestEnv(t, GetCtx(), GetEnv())
 
-	t.Log("checking if dataplane proxy service type changes to ClusterIP")
-	require.Eventually(t, func() bool {
-		servicesClient := GetClients().K8sClient.CoreV1().Services(dataplane.Namespace)
-		dataplaneIngressService, err := servicesClient.Get(GetCtx(), dataplaneIngressService.Name, metav1.GetOptions{})
-		if err != nil {
-			t.Logf("error getting dataplane proxy service: %v", err)
-			return false
-		}
-		if dataplaneIngressService.Spec.Type != corev1.ServiceTypeClusterIP {
-			t.Logf("dataplane proxy service should be of ClusterIP type but is %s", dataplaneIngressService.Spec.Type)
-			return false
-		}
+	t.Log("deploying dataplane resource with 2 replicas")
+	dataplaneName := types.NamespacedName{
+		Namespace: namespace.Name,
+		Name:      uuid.NewString(),
```

Comment:
While we're at it we can change this to use `generateName` (as per #102).

To retain the current references to `dataplaneName` we could use `dataplaneName := client.ObjectKeyFromObject(dataplane)` after it's created.